pkg/api/api.go

```diff
@@ -166,67 +169,80 @@ func doGetFallback(
 	if err != nil {
 		return nil, nil, nil, err
 	}
+
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
 	if !cache {
 		req.Header.Set("Cache-Control", "no-store")
 	}
 
 	resp, data, warnings, err := do(ctx, client, req)
+
 	if resp != nil && resp.StatusCode == http.StatusMethodNotAllowed {
 		u.RawQuery = args.Encode()
 		req, err = http.NewRequest(http.MethodGet, u.String(), nil)
+
 		if err != nil {
 			return nil, nil, warnings, err
 		}
+
 		if !cache {
 			req.Header.Set("Cache-Control", "no-store")
 		}
+
 		return do(ctx, client, req)
 	}
+
 	return resp, data, warnings, err
 }
 
 func QueryRange(ctx context.Context, client promapi.Client, query string, r promapiv1.Range,
-	cache bool) (model.Value, promapiv1.Warnings, error) {
+	cache bool) (model.Value, int, promapiv1.Warnings, error) {
 	u := client.URL(epQueryRange, nil)
 	q := u.Query()
-
 	q.Set("query", query)
 	q.Set("start", formatTime(r.Start))
 	q.Set("end", formatTime(r.End))
 	q.Set("step", strconv.FormatFloat(r.Step.Seconds(), 'f', -1, 64))
 
-	_, data, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, data, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
```

Comment:
I think we should check the `resp` for `nil` here as well.

Reply:
👍